x/ibc/proposal_handler_test.go

```diff
@@ -0,0 +1,162 @@
+package ibc_test
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/suite"
+	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
+	tmtypes "github.com/tendermint/tendermint/types"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/simapp"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/ibc"
+	ibctmtypes "github.com/cosmos/cosmos-sdk/x/ibc/07-tendermint/types"
+	commitmenttypes "github.com/cosmos/cosmos-sdk/x/ibc/23-commitment/types"
+	"github.com/cosmos/cosmos-sdk/x/ibc/types"
+)
+
+var (
+	clientTime = time.Date(2020, 1, 1, 0, 0, 0, 0, time.UTC)
+)
+
+type ProposalHandlerTestSuite struct {
+	suite.Suite
+
+	cdc *codec.LegacyAmino
+	ctx sdk.Context
+	app *simapp.SimApp
+
+	header         ibctmtypes.Header
+	consensusState *ibctmtypes.ConsensusState
+}
+
+func (suite *ProposalHandlerTestSuite) SetupTest() {
+	isCheckTx := false
+	suite.app = simapp.Setup(isCheckTx)
+
+	privVal := tmtypes.NewMockPV()
+	pubKey, err := privVal.GetPubKey()
+	suite.Require().NoError(err)
+
+	val := tmtypes.NewValidator(pubKey, 10)
+	valSet := tmtypes.NewValidatorSet([]*tmtypes.Validator{val})
+
+	suite.header = ibctmtypes.CreateTestHeader(chainID, height+1, height, clientTime.Add(time.Second*5), valSet, valSet, []tmtypes.PrivValidator{privVal})
+
+	suite.cdc = suite.app.LegacyAmino()
+	suite.ctx = suite.app.BaseApp.NewContext(isCheckTx, tmproto.Header{Time: clientTime})
+
+	suite.consensusState = ibctmtypes.NewConsensusState(clientTime, commitmenttypes.NewMerkleRoot([]byte("hash")), height, valSet.Hash())
+
+}
+
+func TestProposalHandlerTestSuite(t *testing.T) {
+	suite.Run(t, new(ProposalHandlerTestSuite))
+}
+
+func testClientUpdateProposal(clientID string, header ibctmtypes.Header) (*types.ClientUpdateProposal, error) {
+	return types.NewClientUpdateProposal("Test", "description", clientID, header)
+}
+
+func (suite *ProposalHandlerTestSuite) testClientState(allowGovernanceOverrideAfterExpire bool, isExpire bool, allowGovernanceOverrideAfterMisbehaviour bool, isFrozen bool) *ibctmtypes.ClientState {
```

Comment:
```suggestion
func (suite *ProposalHandlerTestSuite) testClientState(allowGovernanceOverrideAfterExpire bool, clientTime time.Time, allowGovernanceOverrideAfterMisbehaviour bool, frozenHeight uint64) *ibctmtypes.ClientState {
```

I think this would be more readable